src/services/wizdom.language.update.ts

```diff
@@ -0,0 +1,34 @@
+import { IWizdomWebApiService } from "./webapi/webapi.interfaces";
+import { IWizdomCache } from "./caching/cache.interfaces";
+import { IHttpClient } from "../shared/httpclient.wrappers/http.interfaces";
+
+export class WizdomLanguageUpdate {
+    
+    constructor(private spHttpClient: IHttpClient, private wizdomWebApiService: IWizdomWebApiService, private cache: IWizdomCache) {    }
+
+    public async UpdateIfNeededAsync(webAbsoluteUrl: string, currentUserLanguage: string) { 
+        await this.cache.Localstorage.ExecuteCached("WizdomUserLanguage", async () => {            
+            var apiUrl ="/_api/SP.UserProfiles.PeopleManager/GetMyProperties";
+            return this.spHttpClient.get(webAbsoluteUrl + apiUrl)
```

Comment:
Would'nt it be easier to use await instead of .then here? the function is already declared with async